### DISCUSSION TITLE
subscribe data

### DISCUSSION THREAD

**Discussion by hossein-ghanbari** [TanStack/query#2552] [Q&A] [answered]
Hi .
I wanted some tips on working with the react query library.
How can we detect changes in cached data? Like redux as soon as the data changes. We can have it in all components.
I use queryCache.subscribe but I do not get results. please direct me . Thankful

**Comment by TkDodo** [comment, marked as answer]
that is simply what `useQuery` is doing:

```
const { data } = useQuery(key, queryFn)
```

will subscribe you to this query key, and this component will be informed about changes to the data.

**Reply by hossein-ghanbari** [reply]
By doing this I also request from the server and I do not want to do that and I just want to have a listener for data changes

**Reply by TkDodo** [reply]
the amount of background updates can be customized by setting a `staleTime`

**Reply by dominicfraser** [reply]
I'm wondering if the question being asked here is about an equivalent to https://redux.js.org/api/store#subscribelistener (or this may be a different question).

Redux subscriptions allow scribing to a property deep within the data tree, and doing something when that particular piece of data changes.

I'm not sure if there is a good pattern for this global observer of query cache state, or if it is best to reflect that piece of data into a global store like Redux and subscribe to it there instead?

**Reply by dominicfraser** [reply]
Or maybe this can be used 🤔 
https://react-query.tanstack.com/reference/QueryObserver

**Reply by TkDodo** [reply]
You can use the `select` option on useQuery to subscribe to parts (or properties deep within the data tree) of a query

**Reply by dominicfraser** [reply]
My understanding is that will mean you _only_ subscribe to that part of the query response? Rather than wanting to store the entire response for use in a component, but elsewhere within the application take actions if part of the response was to change?

I realise this is hard to visualise within a full working example 😬

**Reply by TkDodo** [reply]
Yep, that’s right
